src/pt/NovelMania.lua

```diff
@@ -0,0 +1,263 @@
+  "Comédia",
+  "Cotidiano",
+  "Drama",
+  "Ecchi",
+  "Erótico",
+  "Escolar",
+  "Fantasia",
+  "Harém",
+  "Isekai",
+  "Magia",
+  "Mecha",
+  "Medieval",
+  "Militar",
+  "Mistério",
+  "Mitologia",
+  "Psicológico",
+  "Realidade Virtual",
+  "Romance",
+  "Sci-fi",
+  "Sistema de Jogo",
+  "Sobrenatural",
+  "Suspense",
+  "Terror",
+  "Wuxia",
+  "Xianxia",
+  "Xuanhuan",
+  "Yaoi",
+  "Yuri"
+}
+local FILTER_GENRES_VALUES = {
+  [0] = "", -- Todos
+  01, -- Ação
+  02, -- Adulto
+  07, -- Artes Marciais
+  03, -- Aventura
+  04, -- Comédia
+  16, -- Cotidiano
+  23, -- Drama
+  27, -- Ecchi
+  22, -- Erótico
+  13, -- Escolar
+  05, -- Fantasia
+  21, -- Harém
+  30, -- Isekai
+  26, -- Magia
+  08, -- Mecha
+  31, -- Medieval
+  24, -- Militar
+  09, -- Mistério
+  10, -- Mitologia
+  11, -- Psicológico
+  36, -- Realidade Virtual
+  12, -- Romance
+  14, -- Sci-fi
+  15, -- Sistema de Jogo
+  17, -- Sobrenatural
+  29, -- Suspense
+  06, -- Terror
+  18, -- Wuxia
+  19, -- Xianxia
+  20, -- Xuanhuan
+  35, -- Yaoi
+  37 -- Yuri
+}
+
+local FILTER_NATIONALITY_ID = 200
+local FILTER_NATIONALITY_ITEMS = {
+  "Todas",
+  "Americana",
+  "Angolana",
+  "Brasileira",
+  "Chinesa",
+  "Coreana",
+  "Japonesa"
+}
+
+local FILTER_STATUS_ID = 300
+local FILTER_STATUS_ITEMS = { "Todos", "Ativo", "Completo", "Pausado", "Parado" }
+
+local FILTER_ORDERBY_ID = 400
+local FILTER_ORDERBY_ITEMS = {
+  "Qualquer ordem",
+  "Ordem alfabética",
+  "Nº de Capítulos",
+  "Popularidade",
+  "Novidades",
+}
+
+-- ============================== PASSAGE ===============================
+local function shrinkURL(url)
+  return url:gsub("^.-novelmania%.com%.br", "")
+end
+
+local function expandURL(path)
+  return baseURL .. path
+end
+
+--- @param chapterURL string
+--- @return string
+local function getPassage(chapterURL)
+  local document = GETDocument(expandURL(chapterURL))
+  local htmlElement = document:selectFirst("div#chapter-content")
+  -- remove unwanted elements
+  htmlElement:select("h3, div"):remove()
+  return pageOfElem(htmlElement, true)
+end
+
+-- =========================== NOVEL DETAILS ============================
+---@param document Document
+---@return NovelStatus
+local function getStatus(document)
+  local status = document:selectFirst("span.authors:contains(Status:)"):ownText()
+  local status_table = {
+    ["Ativo"] = 0,
+    ["Completo"] = 1,
+    ["Pausado"] = 2
+  }
+  return NovelStatus(status_table[status] or 3)
+end
+
+---@param document Document
+---@return NovelChapter[]
+local function getChapterList(document)
+  local chapters = document:select("ol.list-inline a")
+  local count = 1
+  local chapterList = AsList(map(chapters, function(el)
+    local chapter = NovelChapter {
+      title = el:selectFirst("strong"):text(),
+      link = shrinkURL(el:attr("href")),
+      release = el:selectFirst("small"):text(),
+      order = count
+    }
+    count = count + 1
+    return chapter
+  end))
+  Reverse(chapterList)
```

Comment:
If the chapter list is correctly ordered (i.e. Chapter 1 is at the start of the list), do not Reverse it.